apps/authentication-service/src/shared/auth/authService.ts

```diff
@@ -0,0 +1,112 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+import { TokenSet } from "auth0";
+import { decode, JwtPayload } from "jsonwebtoken";
+
+import envs from "@/config";
+import { IAuthService, ICredentialRequestBody } from "@/shared/auth/authModel";
+import { ISingleton } from "@/shared/models";
+
+export const authEndpoint = "oauth/token";
+
+export const AuthService: ISingleton<IAuthService> = (() => {
+  let instance: IAuthService;
+
+  const createInstance = (): IAuthService => {
+    const domain: string = envs.AUTH0_DOMAIN;
+    const audience: string = envs.AUTH0_AUDIENCE;
+    const clientId: string = envs.AUTH0_CLIENT_ID;
+    const clientSecret: string = envs.AUTH0_CLIENT_SECRET;
+    const tokenFileName: string = envs.TOKEN_CACHE_FILE;
+
+    let token: string;
+    const tokenFile: string = path.join(os.tmpdir(), tokenFileName);
+
+    const setCachedToken = (token: string): void => {
+      try {
+        fs.writeFileSync(tokenFile, token);
+      } catch (Error) {
+        console.log("Unable to cache token", Error);
+      }
+    };
+
+    const getCachedToken = (): string | undefined => {
+      if (fs.existsSync(tokenFile)) {
+        return fs.readFileSync(tokenFile, "utf8");
+      } else {
+        return undefined;
+      }
+    };
+
+    const isTokenExpired = (token: string): boolean => {
```

Comment:
I'd call this getIsTokenExpired. If I read isTokenExpired as a variable name, then I'm going to assume it's a boolean and not a function.